aws-storage-s3/src/test/java/com/amplifyframework/storage/s3/operation/AWSS3StorageDownloadFileOperationTest.kt

```diff
@@ -0,0 +1,105 @@
+import com.amplifyframework.storage.s3.service.StorageService
+import java.io.File
+import org.junit.Before
+import org.junit.Test
+import org.mockito.Mockito
+
+class AWSS3StorageDownloadFileOperationTest {
+
+    private lateinit var awsS3StorageDownloadFileOperation: AWSS3StorageDownloadFileOperation
+    private lateinit var storageService: StorageService
+    private lateinit var cognitoAuthProvider: CognitoAuthProvider
+
+    @Before
+    fun setup() {
+        storageService = Mockito.spy(StorageService::class.java)
+        cognitoAuthProvider = Mockito.mock(CognitoAuthProvider::class.java)
+    }
+
+    @Test
+    fun defaultPrefixResolverAWSS3PluginConfigTest() {
+        val key = "123"
+        val tempFile = File.createTempFile("new", "file.tmp")
+        val expectedKey = "public/123"
+        val request: AWSS3StorageDownloadFileRequest = AWSS3StorageDownloadFileRequest(
+            key,
+            tempFile,
+            StorageAccessLevel.PUBLIC,
+            null
+        )
+        Mockito.`when`(cognitoAuthProvider.identityId).thenReturn("abc")
+        awsS3StorageDownloadFileOperation = AWSS3StorageDownloadFileOperation(
+            storageService,
+            cognitoAuthProvider,
+            request,
+            AWSS3StoragePluginConfiguration(null),
+            {},
+            {},
+            {}
+        )
+        awsS3StorageDownloadFileOperation.start()
+        Mockito.verify(storageService).downloadToFile(expectedKey, tempFile)
+    }
+
+    @Test
+    fun customPrefixResolverAWSS3PluginConfigTest() {
+        val key = "123"
+        val tempFile = File.createTempFile("new", "file.tmp")
+        val expectedKey = "123"
+        val request: AWSS3StorageDownloadFileRequest = AWSS3StorageDownloadFileRequest(
+            key,
+            tempFile,
+            StorageAccessLevel.PUBLIC,
+            null
+        )
+        Mockito.`when`(cognitoAuthProvider.identityId).thenReturn("abc")
+        awsS3StorageDownloadFileOperation = AWSS3StorageDownloadFileOperation(
+            storageService,
+            cognitoAuthProvider,
+            request,
+            AWSS3StoragePluginConfiguration(
+                AWSS3StoragePluginConfiguration.Builder().apply {
+                    awsS3PluginPrefixResolver = object : AWSS3PluginPrefixResolver {
+                        override fun resolvePrefix(
+                            accessLevel: StorageAccessLevel,
+                            targetIdentity: String?,
+                            onSuccess: Consumer<String>,
+                            onError: Consumer<StorageException>
+                        ) {
+                            onSuccess.accept("")
+                        }
+                    }
+                }
+            ),
+            {},
+            {},
+            {}
+        )
+        awsS3StorageDownloadFileOperation.start()
+        Mockito.verify(storageService).downloadToFile(expectedKey, tempFile)
```

Comment:
verifying `""` is good, can we add another test to verify expectedKey == "customPublic/123.png" when the custom prefix resolver returns "customPublic/" for the public access level? Then do we also need to these 3 tests for other operations?